src/main/scala/diplomacy/Nodes.scala

```diff
@@ -668,37 +1486,53 @@ class MixedAdapterNode[DI, UI, EI, BI <: Data, DO, UO, EO, BO <: Data](
   }
 }
 
+/** A node which modifies the parameters flowing through it, but without changing the number of edges or the diplomatic protocol implementation. */
 class AdapterNode[D, U, EO, EI, B <: Data](imp: NodeImp[D, U, EO, EI, B])(
   dFn: D => D,
   uFn: U => U)(
   implicit valName: ValName)
     extends MixedAdapterNode[D, U, EI, B, D, U, EO, B](imp, imp)(dFn, uFn)
 
-// IdentityNodes automatically connect their inputs to outputs
+/** A node which does not modify the parameters nor the protocol for edges that pass through it.
+  *
+  * During hardware generation, identityNodes automatically connect their inputs to outputs.
+  */
 class IdentityNode[D, U, EO, EI, B <: Data](imp: NodeImp[D, U, EO, EI, B])()(implicit valName: ValName)
   extends AdapterNode(imp)({ s => s }, { s => s })
 {
   override def description = "identity"
-  override final def circuitIdentity = true
-  override protected[diplomacy] def instantiate() = {
+  protected override val identity = true
```

Comment:
Yeah, this is of concern. Changing back from `circuitIdentity` to  `identity` is a regression against master 

Reply:
my rebase fault